python/ray/_private/services.py

```diff
@@ -1993,20 +1994,29 @@ def determine_plasma_store_config(
                 )
             else:
                 plasma_directory = ray._private.utils.get_user_temp_dir()
-                logger.warning(
-                    "WARNING: The object store is using {} instead of "
-                    "/dev/shm because /dev/shm has only {} bytes available. "
-                    "This will harm performance! You may be able to free up "
-                    "space by deleting files in /dev/shm. If you are inside a "
-                    "Docker container, you can increase /dev/shm size by "
-                    "passing '--shm-size={:.2f}gb' to 'docker run' (or add it "
-                    "to the run_options list in a Ray cluster config). Make "
-                    "sure to set this to more than 30% of available RAM.".format(
-                        ray._private.utils.get_user_temp_dir(),
-                        shm_avail,
-                        object_store_memory * (1.1) / (2**30),
-                    )
+                recommend_size_gb = object_store_memory * 1.1 / (2**30)
```

Comment:
how is the default `object_store_memory` being determined today @edoakes ? it seems that it will be different value when the machine's memory size is different?

Reply:
is there a way to set a fixed size for `object_store_memory` like via a flag or an env var? this is introducing issues when we run unit tests on different machines or with different parallelism. it will lead to behavior where sometimes it is picking `/dev/shm`, and sometimes it is picking `/tmp`, where `/tmp` is in container is really backed by disks, rather than `tmpfs` in memory.

Reply:
feels that this is ray trying to be too smart ...

Reply:
There is some calculation where I believe it tries to allocate ~30% of the node's memory to the object store by default. The relevant constants are defined here:
https://github.com/ray-project/ray/blob/15727f2059d822ec6918edf179ed13973a5ef326/python/ray/_private/ray_constants.py#L59

This can be overridden using `ray.init(object_store_memory="foo")` or `ray start --object-store-memory <foo>`.

Reply:
This has historically been a cause for a lot of confusion. I somewhat agree that Ray is trying to be too smart here. But at this point it's a behavior that many users' workloads depend on.

Reply:
Look like the default is fixed to be 2gb. Does ray auto-clean up its object on termination? Say if I run 100 tests, each test spawns 1gb of objects, will the latter 98 tests will end up being oom?

Reply:
2gb is only for Mac, not linux (it is 30% of available resources when ray is started by default)

Reply:
>  Say if I run 100 tests, each test spawns 1gb of objects, will the latter 98 tests will end up being oom?

No. 

1. plasma store never OOM. It will spill objects instead. There could be only out of disk space
2. plasma store == huge mmap file. It is auto cleaned up when a raylet terminates